src/applications/vaos/express-care/redux/reducer.js

```diff
@@ -104,11 +105,22 @@ export default function expressCareReducer(state = initialState, action) {
       };
     }
     case FORM_PAGE_CHANGE_COMPLETED: {
+      if (Object.keys(state.newRequest.previousPages).length <= 0) {
+        Object.assign(state.newRequest.previousPages, {
```

Comment:
This is still mutating `previousPages`, but the logic looks good. This is the pattern we tend to use for these kind of updates:

```js
let previousPages = state.newRequest.previousPages;

if (!Object.keys(previousPages).length) {
  previousPages = {
    ...previousPages,
    [action.pageKey]: 'home',
  };
}

if (action.direction === 'next') {
  previousPages = {
     ...previousPages,
     [action.pageKeyNext]: action.pageKey,
  }
}
```

Reply:
Done